packages/angular_devkit/build_ng_packagr/src/build/index.ts

```diff
@@ -12,20 +12,55 @@ import {
   BuilderConfiguration,
   BuilderContext,
 } from '@angular-devkit/architect';
-import { getSystemPath, normalize, resolve } from '@angular-devkit/core';
+import { getSystemPath, normalize, resolve, tags } from '@angular-devkit/core';
+import * as fs from 'fs';
 import * as ngPackagr from 'ng-packagr';
-import { Observable } from 'rxjs';
+import { EMPTY, Observable } from 'rxjs';
+import { catchError, tap } from 'rxjs/operators';
+import * as semver from 'semver';
+
+const NEW_NG_PACKAGR_VERSION = '4.0.0-rc.3';
 
 // TODO move this function to architect or somewhere else where it can be imported from.
 // Blatantly copy-pasted from 'require-project-module.ts'.
 function requireProjectModule(root: string, moduleName: string) {
   return require(require.resolve(moduleName, { paths: [root] }));
 }
 
+function resolveProjectModule(root: string, moduleName: string) {
+  return require.resolve(moduleName, { paths: [root] });
+}
 
 export interface NgPackagrBuilderOptions {
   project: string;
   tsConfig?: string;
+  watch?: boolean;
+}
+
+function checkNgPackagrVersion(projectRoot: string): boolean {
+  let ngPackagrJsonPath;
+
+  try {
+    ngPackagrJsonPath = resolveProjectModule(projectRoot, 'ng-packagr/package.json');
+  } catch {
+    // @angular/service-worker is not installed
```

Comment:
Nit: comment is lying. 

Not really important to get this merged but should be changed if a rebase is needed.

Reply:
I will rebase and do it :)